lib/libzfs_core/libzfs_core.c

```diff
@@ -126,17 +127,20 @@ lzc_ioctl(zfs_ioc_t ioc, const char *name,
 {
 	zfs_cmd_t zc = {"\0"};
 	int error = 0;
-	char *packed;
-	size_t size;
+	char *packed = NULL;
+	size_t size = 0;
 
 	ASSERT3S(g_refcount, >, 0);
 	VERIFY3S(g_fd, !=, -1);
 
-	(void) strlcpy(zc.zc_name, name, sizeof (zc.zc_name));
+	if (name != NULL)
+		(void) strlcpy(zc.zc_name, name, sizeof (zc.zc_name));
```

Comment:
That's right, all the currently existing ioctls provide a name. However, if a future ioctl doesn't need to provide a pool name and gives us NULL we'd panic without this check.
Looks like we should be able to support ioctls without a pool name since we have NO_NAME as a valid value for zfs_ioc_namecheck_t .